src/index.js

```diff
@@ -0,0 +1,73 @@
+import AWS from 'aws-sdk';
+import BPromise from 'bluebird';
+const log = require('saga-logger').create({ module: 'sqs-resource' });
+
+let sqs;
+export const init = (awsConfig = {}, sqsConfig = {}) => {
+  AWS.config.update(awsConfig);
+  sqs = new AWS.SQS(sqsConfig);
+};
+
+export const sendMessage = params => new BPromise((resolve, reject) => {
+  log.debug('SQS_SEND_MESSAGE', null, params);
+
+  sqs.sendMessage(params, (err, data) => {
+    if (err) {
+      log.error('SQS_SEND_MESSAGE_FAIL', err, params);
+      return reject(err);
+    }
+
+    log.debug('SQS_SEND_MESSAGE_SUCCESS', data, params);
+    return resolve(data);
+  });
+});
+
+export const receiveMessage = params => new BPromise((resolve, reject) => {
+  log.debug('SQS_RECEIVE_MESSAGE', null, params);
+
+  sqs.receiveMessage(params, (err, data) => {
+    if (err) {
+      log.error('SQS_RECEIVE_MESSAGE_FAIL', err, params);
+      return reject(err);
+    }
+
+    if (!data.Messages || !data.Messages.length) {
+      log.debug('SQS_RECEIVE_MESSAGE_EMPTY', data, params);
+      return resolve(null);
+    }
+    log.debug('SQS_RECEIVE_MESSAGE_SUCCESS', data, params);
+    return resolve(data.Messages);
+  });
+});
+
+export const pollMessage = params => new BPromise((resolve, reject) => {
+  log.debug('SQS_POLL_MESSAGE', null, params);
+  receiveMessage(params)
+    .then(messages => {
+      log.debug('SQS_POLL_MESSAGE_SUCCESS', messages, params);
+      if (messages) {
+        resolve(messages);
+      } else {
+        log.debug('SQS_POLL_MESSAGE_THROTTLE', { throttling: params.throttling || 1000 }, params);
+        setTimeout(() => pollMessage(params).then(resolve).catch(reject), params.throttling);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 93 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.sagacify.com/coding_rules#rule_key=javascript%3ALineLength)